### PR TITLE
ui: timescale selector bugfixes

### LINKED ISSUE #7590: ui: select timescale sometimes reverts
Sometimes when I select a timescale I see the graphs switch to the new timescale only to revert to the old timescale a few seconds later.

### LINKED ISSUE #7591: ui: select timescale dropdown should dismiss if I click anywhere on the screen
When I have the select timescale dropdown open it should dismiss if I click anywhere else on the screen. Currently you have to click the select timescale button again.

### PR DESCRIPTION
**Pull request by maxlang**
Fixes #7591
Fixes #7590

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8573)

### CODE REVIEW

**Comment by mrtracy** [comment]
LGTM, but could you take a look at the last test in timewindow.spec.tsx?  It was supposed to verify that timeouts were being dealt with correctly, clearly it overlooked this particular bug.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8573#-:-KPJgQmeGukdUx8aDTuy:bftgc99)*

**Comment by maxlang** [comment]
Discussed with @mrtracy and I've decided the difficulty of testing a connected component isn't justified by this change alone. I've created #8595 to track this issue.

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8573#-:-KPK7oz3OJYuXNvqyRUy:b-nfgofg)*

File: ui/app/containers/timescale.tsx [hunk 1]
```diff
@@ -16,22 +16,33 @@ interface TimeScaleSelectorState {
   // This setting should not persist if the current route is changed, and thus
   // it is not stored in the redux state.
   controlsVisible: boolean;
+  // The eventListener is used to track the body event listener which closes
+  // the
```
Comment (tamird) [inline ui/app/containers/timescale.tsx]: ...

Comment (maxlang) [inline ui/app/containers/timescale.tsx]: Whoops, I'll fix this comment in my upcoming PR #8572

**Pull request merged**